src/app.rs

```diff
@@ -255,191 +241,116 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
-
-        Self {
-            init_cfg,
-            cfg: Rc::new(AppCfg {
-                document,
-                mount_point,
-                update,
-                sink,
-                view,
-                window_events,
-                base_path,
-            }),
-            data: Rc::new(AppData {
-                model: RefCell::new(None),
-                // This is filled for the first time in run()
-                main_el_vdom: RefCell::new(None),
-                popstate_closure: RefCell::new(None),
-                hashchange_closure: RefCell::new(None),
-                routes: RefCell::new(routes),
-                window_event_handler_manager: RefCell::new(EventHandlerManager::new()),
-                sub_manager: RefCell::new(SubManager::new()),
-                msg_listeners: RefCell::new(Vec::new()),
-                scheduled_render_handle: RefCell::new(None),
-                after_next_render_callbacks: RefCell::new(Vec::new()),
-                render_info: Cell::new(None),
-            }),
-        }
-    }
-
     /// Bootstrap the dom with the vdom by taking over all children of the mount point and
     /// replacing them with the vdom if requested. Will otherwise ignore the original children of
     /// the mount point.
-    fn bootstrap_vdom(&self, mount_type: MountType) -> El<Ms> {
+    fn bootstrap_vdom(&self) -> El<Ms> {
         // "new" name is for consistency with `update` function.
         // this section parent is a placeholder, so we can iterate over children
         // in a way consistent with patching code.
         let mut new = El::empty(Tag::Placeholder);
 
         // Map the DOM's elements onto the virtual DOM if requested to takeover.
-        if mount_type == MountType::Takeover {
-            // Construct a vdom from the root element. Subsequently strip the workspace so that we
-            // can recreate it later - this is a kind of simple way to avoid missing nodes (but
-            // not entirely correct).
-            // TODO: 1) Please refer to [issue #277](https://github.com/seed-rs/seed/issues/277)
-            let mut dom_nodes: El<Ms> = (&self.cfg.mount_point).into();
-            #[cfg(debug_assertions)]
-            dom_nodes.warn_about_script_tags();
-
-            dom_nodes.strip_ws_nodes_from_self_and_children();
-
-            // Replace the root dom with a placeholder tag and move the children from the root element
-            // to the newly created root. Uses `Placeholder` to mimic update logic.
-            new.children = dom_nodes.children;
+        // Construct a vdom from the root element. Subsequently strip the workspace so that we
+        // can recreate it later - this is a kind of simple way to avoid missing nodes (but
+        // not entirely correct).
+        // TODO: 1) Please refer to [issue #277](https://github.com/seed-rs/seed/issues/277)
+        let mut dom_nodes: El<Ms> = (self.cfg().mount_point()).into();
```

Comment:
accessing cfg and mount_point field with a getter method was to help with development. should be trivial to revert

Reply:
Not sure how familiar you are with Rust, but you should probably make them either `pub(super)` or `pub(crate)` fields instead. Getters and setters aren't usually used in Rust, unless there's some specific state you're maintaining due to a few reasons. (I don't think this is the case for your getter/setter, but correct me if I'm wrong.) [Here's a note on the topic, provided courtesy of r/rust (Reddit).](https://www.reddit.com/r/rust/comments/6etrr1/a_derive_for_your_basic_getters_and_setters/did5zoh?utm_source=share&utm_medium=web2x).

If you really want to create something like a "getter"/"setter", it's more useful to create "view" objects. Not going to go into them here, but just a note.

If you were trying to highlight where `cfg` and `mount_point` were used, though, then it's fine. But please revert the getters afterwards.

Reply:
This was done intentionally in order to add more constraints to the library internals. That way, I could be more confident about the changes I was making. I'll reverse it.